versioned/persist/inmem/src/main/java/org/projectnessie/versioned/persist/inmem/InmemoryDatabaseAdapter.java

```diff
@@ -252,4 +257,69 @@ protected RefLog fetchFromRefLog(NonTransactionalOperationContext ctx, Hash refL
         .map(ProtoSerialization::protoToRefLog)
         .collect(Collectors.toList());
   }
+
+  @Override
+  protected void writeAttachments(Stream<Entry<AttachmentKey, AttachmentValue>> attachments) {
+    attachments.forEach(
+        e -> store.attachments.put(dbKey(e.getKey().toByteString()), e.getValue().toByteString()));
+  }
+
+  @Override
+  protected boolean consistentWriteAttachment(
+      AttachmentKey key, AttachmentValue value, Optional<String> expectedVersion) {
+    try {
+      store.attachments.compute(
+          dbKey(key.toByteString()),
+          (k, current) -> {
+            if (expectedVersion.isPresent()) {
+              if (current != null) {
+                try {
+                  AttachmentValue currentValue = AttachmentValue.parseFrom(current);
+                  if (currentValue.hasVersion()
+                      && currentValue.getVersion().equals(expectedVersion.get())) {
+                    return value.toByteString();
+                  }
+                } catch (InvalidProtocolBufferException e) {
+                  throw new RuntimeException(e);
```

Comment:
shouldn't we create a custom exception that extends `RuntimeException` instead of using it directly?

Reply:
Generally, yes.
But this is the inmemory implementation, that's only used in tests (hope nobody uses it in prod...)

Reply:
maybe add it in the Javadoc of this class (that it should not be used in prod)?

Reply:
Different topic ;)
If, then it should be everywhere - especially application.properties, website, etc. Wanna take a stab on that?

Reply:
Maybe we should create a ticket for that and decide if it is worth doing depending on our priorities?

Reply:
Yup, please go ahead.

Reply:
created: https://github.com/projectnessie/nessie/issues/3414